test/testlib/minikube_utilities.go

```diff
@@ -29,21 +30,86 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+/** Lists of the teardown and diagnostic teardown funcs */
 var teardownLists = make(map[string][]func())
+var diagnosticTeardownLists = make(map[string][]func())
+
+/** Exported var - initialised from the EnvVar, but can be reset in code if desired */
+var AlwaysRunDiagnosticTeardowns = strings.EqualFold(os.Getenv(ALWAYS_RUN_DIAGNOSTIC_TEARDOWNS), "true")
```

Comment:
Do we set this to `true` somewhere in `.travis.yml` file or the intuition is to overwrite it only when running tests locally?

Reply:
the idea is to set it locally if you need it, yeah

Reply:
To complete the response - the idea is that the diagnostic teardowns are *not* run in Travis by default.

To have a diagnostic teardown run in Travis, you should pass in a 'conditional' that will be true when the diagnostic teardown should be run - for example:

* t.Failed() - will run if the test has already been marked as failed (the t.Failed() test is run at Teaardown() time)
* true - will run unconditionally - even if ALWAYS_RUN_DIAGNOSTIC_TEARDOWNS = false